cap/01-categorie.tex

```diff
@@ -130,7 +150,86 @@ \subsubsection*{Esercizi}
+  \item In $\ctVect_\ictR$, gli isomorfismi sono le biiezioni lineari. In particulare, in dimensione finita, due spazi vettoriali sono isomorfi se e solo se hanno la stessa dimensione.
+  \item In $\ctTop$, gli isomorfismi sono gli omeomorfismi: funzioni continue con un'inversa continua. 
+  \item In $\ctGrp$, gli isomorfismi sono gli isomorfismi di gruppi. 
+ \end{itemize}
+\end{examples}
+
+\begin{warning}
+ Spesso una categoria viene chiamata con il nome dei suoi oggetti ($\ctVect$, $\ctGrp$, eccetera), ma è importante sapere \emph{quali sono i morfismi} per sapere che struttura in particolare si sta studiando. Questo è particolarmente importante nel caso degli isomorfismi: per esempio, $\ictR$ e $\ictR^2$, come insiemi, sono isomorfi, ma non come spazi vettoriali. Per cui, nella categoria $\ctVect$, gli oggetti $\ictR$ e $\ictR^2$ sono isomorfi, ma non sono isomorfi se li intendiamo come oggetti della categoria $\ctSet$. 
+\end{warning}
+
+Si noti che le condizioni $g\circ f = \id_X$ e $f\circ g=\id_Y$ servono entrambe, perché possono valere o non valere indipendentemente l'una dall'altra. 
+\begin{example}
+ Nella categoria $\ctVect_\ictR$, si consideri la mappa $f:\ictR\to\ictR^2$ data dall'inclusione dell'asse $X$, $x\mapsto(x,0)$. Data la proiezione sull'asse $X$ data da $g:\ictR^2\to\ictR$, $(x,y)\mapsto x$, si ha che $g\circ f=\id_\ictR$. Però non è vero che $f\circ g = \id_{\ictR^2}$. (E infatti i due oggetti non sono isomorfi come spazi vettoriali.)
+\end{example}
+
+\begin{definition}
+ Una categoria in cui tutti i morfismi sono isomorfismi si chiama \emph{grouppoide}.\footnote{In inglese: \emph{groupoid}.} 
+\end{definition}
+
+\begin{example}
+ Abbiamo visto che un preordine (una relazione riflessiva e transitiva) si può vedere come una categoria. Questa categoria è un gruppoide se e solo se la relazione è anche simmetrica (cioè, è un'equivalenza). Se $x\sim y$, abbiamo un unico morfismo $x\to y$. Questo morfismo è invertibile se e solo se esiste un morfismo $y\to x$, cioè, se anche $y\sim x$. (Si noti che non ci sono altre condizioni da soddisfare per avere un inverso: per esempio, il morfismo composto $x\to y\to x$ è necessariamente uguale a $\id_x$ per unicità.)
+\end{example}
+
+\begin{example}
+ Abbiamo visto che ogni monoide $M$ si può considerare una categoria con un solo oggetto. Questa categoria è un gruppoide se e solo se $M$ è un gruppo. Infatti, un morfismo $m$ ammette un inverso nel senso della teoria delle categorie se e solo se ammette un inverso nel senso della teoria dei gruppi: $g\cdot g^{-1}=g^{-1}\cdot g = e$. 
+\end{example}
+
+Un \emph{automorfismo} è un endomorfismo (un morfismo da un oggetto a sé stesso) invertibile. In un gruppo, tutti i morfismi sono automorfismi.
+
+\begin{example}
+ La categoria i cui oggetti sono gli insiemi e i cui morfismi sono biiezioni è un gruppoide.
+\end{example}
+
```

Comment:
Mi piacerebbe aggiungere una frase come <<Quando si prendono solo gli insiemi finiti, questa categoria diventa molto interessante! E' "l'unione disgiunta" (in un senso che preciseremo in \ref{}) di tutti i gruppi simmetrici S_n>>